### PR TITLE
feat: add slog JSON handler

### PR DESCRIPTION
**Pull request by Kalachevskyi**

### CODE REVIEW

File: handlers/slog/doc.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Package sloghandler provides a configurable logging handler for the Slog library, allowing users
+// to customize various aspects of log handling, including output destination, log level, attribute
+// replacement, and source information addition.
+package sloghandler
```
Comment (vpakhuchyi) [inline handlers/slog/doc.go:61]: I would add an example of the slog initialization with this `sloghandler`.

File: handlers/slog/handler.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	cfg.HandlerOptions.ReplaceAttr = func(groups []string, a slog.Attr) slog.Attr {
+		attr := a
+		if cfg.ReplaceAttr != nil {
+			attr = cfg.ReplaceAttr(groups, a)
+		}
+
+		switch attr.Key {
+		case slog.TimeKey, slog.LevelKey, slog.SourceKey:
```
Comment (vpakhuchyi) [inline handlers/slog/handler.go:43]: I would add a comment with an explanation of why Censor execution is not required for these keys.

File: handlers/slog/options.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"io"
+	"log/slog"
+
+	"github.com/vpakhuchyi/censor"
+)
+
+// Option represents a set of options for configuring the Slog Handler along with a censor processor.
+// It is used as arguments for the NewJSONHandler function. See NewJSONHandler for more details.
+type Option func(options *config)
```
Comment (vpakhuchyi) [inline handlers/slog/options.go]: I think the current naming in this signature may confuse a bit. 
`type Option func(options *config)` - so i expect to see a func that accepts options but it accepts a config.
What do you think about changing it to `type Option func(cfg *config)`?

File: handlers/slog/options.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package sloghandler
+
+import (
+	"io"
+	"log/slog"
+
+	"github.com/vpakhuchyi/censor"
+)
+
+// Option represents a set of options for configuring the Slog Handler along with a censor processor.
```
Comment (vpakhuchyi) [inline handlers/slog/options.go]: It feels like the description isn't correct. Based on the implementation. Option is just an option, not a set (or a few of them).
I mean, each option may apply only the only characteristic/action.

File: handlers/slog/doc.go (hunk 1 above)
Comment (Kalachevskyi) [reply handlers/slog/doc.go:61]: Make sense, thank you

File: handlers/slog/handler.go (hunk 2 above)
Comment (Kalachevskyi) [reply handlers/slog/handler.go:43]: Fixed, thanks

File: handlers/slog/options.go (hunk 3 above)
Comment (Kalachevskyi) [reply handlers/slog/options.go]: Make sense, thank you

File: handlers/slog/options.go (hunk 4 above)
Comment (Kalachevskyi) [reply handlers/slog/options.go]: Make sense, thank you

**Review by vpakhuchyi — APPROVED** [review]

**Pull request merged**